src/main/java/com/ss/training/lms/controller/AdminPublisherController.java

```diff
@@ -38,5 +40,27 @@
 		}
 		return new ResponseEntity<Publisher>(publisher, status);
 	}
-	
+
+	/**
+	 * @param publisher
+	 * @return
+	 */
+	@PutMapping(path = "/lms/admin/publisher")
+	public ResponseEntity<Publisher> updatePublisher(@RequestBody Publisher publisher) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if (publisher == null || publisher.getPublisherName() == null || publisher.getPublisherName().length() > 45
+				|| (publisher.getAddress() != null && publisher.getAddress().length() > 45)
+				|| (publisher.getPhone() != null && publisher.getPhone().length() > 45))

```

Comment:
Do we need to check the length here since the database has that rule set.

Reply:
As with the author, it allows us to distinguish between client and server errors. The service will throw SQLException if there was some problem with a database operation, whether the problem was a string being too long, a connection configuration error, or something else.